adhocteam/Head-Start-TTADP
# Authors can resubmit a report for approval

**Description of change**

Backend changes:
 * Report statuses are now an enum in the DB and constants in the backend
 * Activity reports now include the full approving manager instead of only `approvingManagerId`

Frontend changes:
 * Report statuses are now constants that match enum in backend
 * Review page now branches on the report status in addition to author vs approving manager
 * Report with no page set now opens to activity summary if the status is "draft" but opens to the review page otherwise

**How to test**

NOTE: Testing this out fully requires switching users from the author to the approving manager

1) Pull down
2) Give a user approve permissions on region 1 (using the `admin` page), note the user id, this is the approving manager
3) Fill out a report, submit for approval from the approving manager from step 2
4) Restart your server with `CURRENT_USER_ID` set to the approving manager id from step 2 (or edit `src/middleware/authMiddelware.js` line 47)
5) Load the activity report created in step 3. You should be redirected to the review page and see the approving manager view of the review page
6) Mark the report as `Needs Action`
7) Switch back to the author (should be id 1, Hermione), load up the report
8) You should be on the re-submit page, go ahead and re-submit
9) Switch back to the approving manager and approve the report, you should see the "Report approved" page
10) If you are still with me, you should see the same "Report approved" page if you switch back to the author

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/225

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
